QModManager/Utility/IOUtilities.cs

```diff
@@ -0,0 +1,180 @@
+                {
+                    toWrite += GetFolderStructureRecursively(dir, spaces + 4);
+                }
+
+                string[] files = Directory.GetFiles(directory);
+                for (int i = 1; i <= files.Length; i++)
+                {
+                    FileInfo fileinfo = new FileInfo(files[i - 1]);
+                    if (i != files.Length)
+                        toWrite += $"{GenerateSpaces(spaces + 4)}|---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                    else
+                        toWrite += $"{GenerateSpaces(spaces + 4)}`---- {fileinfo.Name} ({ParseSize(fileinfo.Length)})\n";
+                }
+
+                return toWrite;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        private static string ParseSize(long lsize)
+        {
+            float size = lsize;
+            string unit;
+
+            if (size < 1024)
+            {
+                unit = "B";
+            }
+            else
+            {
+                size /= 1024;
+                if (size < 1024)
+                {
+                    unit = "KB";
+                }
+                else
+                {
+                    size /= 1024;
+                    if (size < 1024)
+                    {
+                        unit = "MB";
+                    }
+                    else
+                    {
+                        size /= 1024;
+                        if (size < 1024)
+                        {
+                            unit = "GB";
+                        }
+                        else
+                        {
+                            size /= 1024;
+                            unit = "TB";
+                        }
+                    }
+                }
+            }
+
+            string number = size.ToTwoDecimalString();
+            number.TrimEnd('0');
+            number.TrimEnd('.');
+
+            return number + unit;
+        }
+        private static long CalculateFolderSizeRecursively(string directory)
+        {
+            try
+            {
+                long size = 0;
+                foreach (string dir in Directory.GetDirectories(directory))
+                {
+                    size += CalculateFolderSizeRecursively(dir);
+                }
+                foreach (string file in Directory.GetFiles(directory))
+                {
+                    size += new FileInfo(file).Length;
+                }
+                return size;
+            }
+            catch (Exception e)
+            {
+                throw e;
+            }
+        }
+
+        private static string GenerateSpaces(int spaces)
+        {
+            string s = "";
+            for (int i = 1; i <= spaces; i+=4)
+                s += "|   ";
+            return s;
+        }
+
+        private static int GetFileCountRecursively(string directory)
```

Comment:
File counts also seems irrelevant for what this could be used for.  
Is it really needed?

Reply:
Not really, I just think it's "fancy"